rest_framework/serializers.py

```diff
@@ -1076,17 +1073,30 @@ def get_fields(self):
 
         return fields
 
+    def get_fields(self):
+        """
+        Return the dict of field names -> field instances that should be
+        used for `self.fields` when instantiating the serializer.
+        """
+        cls = self.__class__
+        # We don't want the cache to traverse the MRO, since each subclass
+        # has its own fields; hence the cls comparison.
+        if not hasattr(cls, "_fields_cache") or cls._fields_cache[0] != cls:
+            cls._fields_cache = cls, cls._get_fields()
+        return copy.deepcopy(cls._fields_cache[1])
```

Comment:
Two thoughts:
- I would prefer to mimic model forms and use `base_fields` instead of `_fields_cache. The class itself can always be stored in a separate attribute.
- If we were to use a metaclass to call `_get_fields`, caching the class wouldn't be necessary.